crates/ruff_linter/src/rules/pylint/rules/no_self_use.rs

```diff
@@ -105,11 +110,28 @@ pub(crate) fn no_self_use(checker: &Checker, scope: &Scope, diagnostics: &mut Ve
         return;
     };
 
-    if parameter.name.as_str() == "self"
-        && scope
-            .get("self")
-            .map(|binding_id| checker.semantic().binding(binding_id))
-            .is_some_and(|binding| binding.kind.is_argument() && !binding.is_used())
+    if parameter.name.as_str() != "self" {
+        return;
+    }
+
+    // If the method contains a `super` reference, then it should be considered to use self
+    // implicitly.
+    if let Some(binding_id) = checker.semantic().global_scope().get("super") {
+        let binding = checker.semantic().binding(binding_id);
+        if binding.kind.is_builtin() {
+            if binding
+                .references()
+                .any(|id| checker.semantic().reference(id).scope_id() == scope_id)
+            {
+                return;
+            }
+        }
+    }
```

Comment:
@LaBatata101 - What do you think of this?

Reply:
LGTM. Didn't know about the `binding.references()` function, will be very helpful when implementing other rules in the future.